mountpoint-s3-client/src/s3_crt_client.rs

```diff
@@ -449,6 +449,7 @@ impl S3CrtClientInner {
 
                 let http_status = metrics.status_code().unwrap_or(-1);
```

Comment:
Should we leave this as Option, rather than introduce the confusing `-1` status?

Reply:
Yeah, I think that makes sense. I had the same thoughts, wasn't sure whether to leave it as is (since `-1` can have its own convention outside of Rust) but maybe not worth it.

I'm thinking we switch to debug anyway, so let's just go with consistency at this point. I'll change to `None`.

Reply:
I'd still be keeping it as `-1` in the failure metric at the bottom for now.